### ISSUE TITLE
get_uint_info - RuntimeError: expected float pointer

### ISSUE THREAD

**Issue by KOLANICH** [dmlc/xgboost#2967]
When I call ```get_uint_info("root_index")``` I get the error ```RuntimeError: expected float pointer```. It seems that python API doesn't have it implemented right, since it assumes it to be always float.

The fix is:

```python
npToPointerTMapping={
	np.float32: ctypes.c_float,
	np.uint32: ctypes.c_uint,
}
def ctypes2numpy(cptr, length, dtype):
    """Convert a ctypes pointer array to a numpy array.
    """
    if dtype not in npToPointerTMapping:
        raise RuntimeError('Supported  types:', list(npToPointerTMapping.keys()))
    if not isinstance(cptr, ctypes.POINTER(npToPointerTMapping[dtype])):
        raise RuntimeError('expected '+str(npToPointerTMapping[dtype])+' pointer')
    res = np.zeros(length, dtype=dtype)
    if not ctypes.memmove(res.ctypes.data, cptr, length * res.strides[0]):
        raise RuntimeError('memmove failed')
    return res
```

**Comment by KOLANICH** [comment]
@tqchen, is it fixed?

**Comment by hcho3** [comment]
Addressed in #3442.

**Comment by KOLANICH** [comment]
Thank you.
